space/space.go

```diff
@@ -301,7 +301,15 @@ func (r *GormRepository) listSpaceFromDB(ctx context.Context, q *string, userID
 	}
 
 	// ensure that the result list is always ordered in the same manner
-	db = db.Order("spaces.updated_at DESC")
+	// if no sorting info provided
+	if sort == nil {
+		db = db.Order("spaces.updated_at DESC")
+	} else if *sort == "name" {
+		db = db.Order("spaces.name ASC")
+	} else if *sort == "ownerID" {
+		db = db.Order("spaces.owner_id ASC")
+	}
```

Comment:
IMO, this should go to controller and repository method should just perform the main operation.
```
const (
	SortByUpdatedAt string = "updated_at"
	SortByName string = "name"
)
```
then in the controller
```
if ctx.Sort == nil {
 ctx.Sort = SortByUpdatedAt
}else {
 ctx.Sort = SortByName
}
```
and pass `*ctx.Sort` to the repository method.
So, you can just do `db = db.Order(sort)`

Reply:
done